crates/full-node/sov-db/src/storage_manager/nomt_based/groups.rs

```diff
@@ -5,23 +5,123 @@ use std::thread::JoinHandle;
 
 use anyhow::Context;
 use rockbound::cache::delta_reader::DeltaReader;
-use rockbound::SchemaBatch;
+use rockbound::rocksdb::ColumnFamilyDescriptor;
+use rockbound::versioned_db::{VersionedDB, VersionedDeltaReader, VersionedTableMetadataKey};
+use rockbound::{default_cf_descriptor, SchemaBatch};
 use sov_rollup_interface::reexports::digest;
 
 use crate::accessory_db::AccessoryDb;
 use crate::config::RollupDbConfig;
-use crate::historical_state::HistoricalStateReader;
+use crate::historical_state::{HistoricalStateReader, StateChanges};
 use crate::ledger_db::LedgerDb;
+use crate::metrics::nomt::PrunerMetric;
 use crate::namespaces::{KernelNamespace, UserNamespace};
 use crate::pruner::Pruner;
-use crate::schema::namespace::StateValues;
-use crate::schema::tables::ModuleAccessoryState;
+use crate::schema::namespace::{
+    NomtCommittedVersion, NomtHistoricalState, NomtPruningState, NomtStateValues,
+};
+use crate::schema::tables::{ModuleAccessoryState, StateRootHashes};
 use crate::state_db_nomt::{NomtSessionBuilder, NomtStateDb, StateOverlay};
 use crate::storage_manager::{update_ledger_finalized_height, InitializableNativeNomtStorage};
+use crate::DbOptions;
+
+/// A database to store the flat state of the rollup (i.e. the raw key-value pairs)
+pub struct FlatStateDb {
```

Comment:
Nit: Could we place this struct into top of `sov-db` crate, so it is similar to what is already there, plus `storage_manger` module contains only storage manager related code

Reply:
Done